TypeScript compile error in @middy/core 5.2.4

**Describe the bug**
After updating @middy/core from 5.2.3 to 5.2.4, an error occurred in TypeScript compile (tsc).

**To Reproduce**
How to reproduce the behaviour:

Sample code 

- packages/server/src/lambda/handlers/api-gateway/rest-api/router.ts

```ts
import serverlessExpress from "@codegenie/serverless-express";
import middy from "@middy/core";
import express from "express";

const app = express();

export const handler = middy(serverlessExpress({ app }));
```

Thrown error

```shell
tsc --noEmit

src/lambda/handlers/api-gateway/rest-api/router.ts:11:30 - error TS2345: Argument of type 'Handler<any, any> & ConfigureResult<any, any>' is not assignable to parameter of type 'PluginObject | MiddyInputHandler<any, any, Context> | undefined'.
  Type 'Handler<any, any> & ConfigureResult<any, any>' is not assignable to type 'MiddyInputHandler<any, any, Context>'.
    Types of parameters 'callback' and 'opts' are incompatible.
      Type 'MiddyHandlerObject' is not assignable to type 'Callback<any>'.
        Type 'MiddyHandlerObject' provides no match for the signature '(error?: string | Error | null | undefined, result?: any): void'.

11 export const handler = middy(serverlessExpress({ app }));
```

**Expected behaviour**
tsc can run without errors.

**Environment (please complete the following information):**

- Node.js: 20.9.0
- Middy: 5.2.4
- AWS SDK 2.1450.0

--- a/package.json
+++ b/package.json
@@ -70,7 +70,8 @@
     "speedscope": "^1.0.0",
     "tinybench": "^2.5.1",
     "ts-standard": "^12.0.0",
-    "tsd": "^0.30.0"
+    "tsd": "^0.30.0",
+    "typescript": "^5.4.3"
   },
   "workspaces": [
     "packages/*"

--- a/packages/core/index.d.ts
+++ b/packages/core/index.d.ts
@@ -168,6 +168,7 @@ declare function middy<
   TInternal extends Record<string, unknown> = {}
 > (
   handler?:
+  | LambdaHandler<TEvent, TResult>
   | MiddlewareHandler<LambdaHandler<TEvent, TResult>, TContext>
   | PluginObject,
   plugin?: PluginObject

--- a/packages/core/index.test-d.ts
+++ b/packages/core/index.test-d.ts
@@ -1,5 +1,5 @@
 import { expectType, expectAssignable } from 'tsd'
-import middy from '.'
+import middy, { MiddyfiedHandler } from '.'
 import {
   APIGatewayProxyEvent,
   APIGatewayProxyResult,
@@ -64,7 +64,7 @@ expectType<Handler>(handler)
 expectAssignable<AWSLambdaHandler<APIGatewayProxyEvent, APIGatewayProxyResult>>(handler)
 
 // Middy handlers third argument is an object containing a abort signal
-middy((event, context, { signal }) => expectType<AbortSignal>(signal))
+middy((_event: any, _context: any, { signal }: { signal: AbortSignal }) => expectType<AbortSignal>(signal))
 
 // invokes the handler to test that it is callable
 // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
@@ -392,3 +392,11 @@ expectType<middy.MiddyfiedHandler<unknown>>(syncedStreamifiedResponseHandler)
 
 syncedStreamifiedResponseHandler.handler(syncedLambdaHandler)
 syncedStreamifiedResponseHandler.use(middlewareObj)
+
+// Issue #1176
+const baseHandler: AWSLambdaHandler = async (event) => {
+  console.log('Hello world')
+}
+
+const handler1176 = middy(baseHandler)
+expectType<MiddyfiedHandler<any, any, Error, Context, {}>>(handler1176)

